File: TypeScript/Array and Strings/Easy/findMissingNumber.ts
function findMissingNumber(nums: number[]): number {
    const len: number = nums.length;

    const max: number = Math.max(...nums);

    // we calculate the sum of all numbers from one to the maximum
    // number in the array
    let sum = (max * (max + 1)) / 2;

    for (let i: number = 0; i < len; i++) {
        sum -= nums[i];
    }

    return sum;
}

console.log(findMissingNumber([2, 4, 1, 6, 7, 5]));

File: C/Arrays/find_missing_number.c
#include <stdio.h>

int find_missing_number(int *arr, int size) {
    // because we are missing one number so the size will be less by 1
    int n = size + 1;

    // we apply this mathematical formula to find the sum of the numbers from 1 to n
    int sum = (n * (n + 1)) / 2;

    for (int i = 0; i < size; i++) {
        sum -= arr[i];
    }

    return sum;
}

int main() {
    int arr[] = {1, 2, 3, 5, 6};
    int size = 5;

    printf("%d\n", find_missing_number(arr, size));
    return 0;
}